eclair-node/src/main/scala/fr/acinq/eclair/Boot.scala

```diff
@@ -84,6 +97,46 @@ object Boot extends App with Logging {
     }
   }
 
+  /**
+   * Generates the cookie file and return the password
+   */
+  def generateCookie(pathString: String, groupReadable: Boolean): String = {
+    val hexPassword = randomBytes32().toHex
+
+    val path = Path.of(pathString)
+    Files.deleteIfExists(path)
+    Files.createFile(path)
+
+    val posixView = Files.getFileAttributeView(path, classOf[PosixFileAttributeView])
+    val aclView = Files.getFileAttributeView(path, classOf[AclFileAttributeView])
+    if (posixView != null) {
+      // Change permissions for the .cookie file on Linux/Mac OS
+      if (groupReadable) {
+        posixView.setPermissions(PosixFilePermissions.fromString("rw-r-----"))
+      } else {
+        posixView.setPermissions(PosixFilePermissions.fromString("rw-------"))
```

Comment:
Do you need to change it in that case? What are the default permissions if you don't do anything?